kratos/solving_strategies/strategies/solving_strategy.h

```diff
@@ -341,20 +342,14 @@ class SolvingStrategy
     {
         KRATOS_TRY
 
-        KRATOS_ERROR_IF(GetModelPart().NodesBegin()->SolutionStepsDataHas(DISPLACEMENT_X) == false) << "It is impossible to move the mesh since the DISPLACEMENT var is not in the Model Part. Either use SetMoveMeshFlag(False) or add DISPLACEMENT to the list of variables" << std::endl;
+        KRATOS_ERROR_IF_NOT(GetModelPart().HasNodalSolutionStepVariable(DISPLACEMENT_X)) << "It is impossible to move the mesh since the DISPLACEMENT var is not in the Model Part. Either use SetMoveMeshFlag(False) or add DISPLACEMENT to the list of variables" << std::endl;
 
-        NodesArrayType& NodesArray = GetModelPart().Nodes();
-        const int numNodes = static_cast<int>(NodesArray.size());
+        block_for_each(GetModelPart().Nodes(), [](Node<3> rNode){
```

Comment:
ouch, here you create a copy

```suggestion
        block_for_each(GetModelPart().Nodes(), [](Node<3>& rNode){
```

Reply:
yes I noticed already :man_facepalming: